catalog/transformer/catalog-transformer-csv-queryresponse/src/main/java/ddf/catalog/transformer/csv/CsvMetacardTransformer.java

```diff
@@ -43,11 +46,20 @@ public BinaryContent transform(Metacard metacard, Map<String, Serializable> argu
 
     Map<String, String> aliases =
         (Map<String, String>) arguments.getOrDefault("aliases", new HashMap<>());
+    String attributeString = (String) arguments.get("columnOrder");
+    ArrayList<String> attributes =
+        new ArrayList<String>(Arrays.asList((attributeString).split(",")));
 
     Appendable appendable =
         writeMetacardsToCsv(
             Collections.singletonList(metacard),
-            new ArrayList<>(metacard.getMetacardType().getAttributeDescriptors()),
+            new ArrayList<AttributeDescriptor>(
+                metacard
+                    .getMetacardType()
+                    .getAttributeDescriptors()
+                    .stream()
+                    .filter(attr -> attributes.contains(attr.getName()))
```

Comment:
Is this transformer used anywhere else? wondering if we should avoid filtering if no `columnOrder` arg is passed in

Reply:
I don't think it's used anywhere else but it's probably a good idea to add that logic anyway